lib/puppet/resource/capability_finder.rb

```diff
@@ -22,6 +26,29 @@ def self.find(environment, code_id, cap)
       raise Puppet::DevError, 'PuppetDB is not available'
     end
 
+    resources = search(environment, nil, cap)
+
+    if resources.size > 1 && code_id
+      Puppet.debug "Found multiple resources when looking up capability #{cap}, filtering by code id #{code_id}"
+      resources = search(environment, code_id, cap)
+    end
+
+    if resources.size > 1
+      raise Puppet::DevError,
+        "Unexpected response from PuppetDB when looking up #{cap}:\n" +
+        "expected exactly one resource but got #{resources.size};\n" +
+        "returned data is:\n#{resources.inspect}"
+    end
+
+    if resource_hash = resources.first
+      instantiate_resource(resource_hash)
+    else
+      Puppet.debug "Could not find capability resource #{cap} in PuppetDB"
+      nil
```

Comment:
Without the `nil`, it will return whatever the return value of `Puppet.debug` is. I think that that's only `nil` if you're not running with debug level logging. If you have debug on, it seems like it will return a Log object. In any case, relying on the return value of `Puppet.debug` seems bad.